### PR TITLE
Add type hints to core.py

### PR DESCRIPTION
**Pull request by zachschillaci27**
Hi all,

Just a quick PR to add type hinting to `core.py`

I noticed a few methods are not currently used in the codebase, such as `build_bert_batch_from_txt` and `listify_batch`. Not sure if they are being used elsewhere or we should clean them up. Let me know!

Zach

### CODE REVIEW

**Comment by lvwerra** [comment]
Hi @zachschillaci27, thanks for helping cleanup. Indeed those functions are not used anymore and can be removed!

**Comment by zachschillaci27** [comment]
> Hi @zachschillaci27, thanks for helping cleanup. Indeed those functions are not used anymore and can be removed!

Got it, done!

**Review by lvwerra — APPROVED** [review]

**Comment by lvwerra** [comment]
Could you run `make precommit`  (requires `pre-commit` to be installed) to fix the code quality checks?

File: trl/core.py [hunk 1]
```diff
@@ -188,37 +189,9 @@ def stats_to_np(stats_dict):
     return new_dict
 
 
-def listify_batch(tensor):
-    """Turns the first dimension of a tensor into a list."""
-    return [tensor[i] for i in range(tensor.shape[0])]
-
-
-def build_bert_batch_from_txt(text_list, tokenizer, device):
-    """Create token id and attention mask tensors from text list for BERT classification."""
```
Comment (younesbelkada) [inline trl/core.py:197]: Nice catch, these methods are not used at all in TRL

**Review by younesbelkada — APPROVED** [review]
Great clean up ! Thanks !

File: trl/core.py (hunk 1 above)
Comment (zachschillaci27) [reply trl/core.py:197]: Happy to help!

**Pull request merged**